cs.c

```diff
@@ -557,9 +574,14 @@ static void fill_insn(struct cs_struct *handle, cs_insn *insn, char *buffer, MCI
 		struct insn_mnem *tmp = handle->mnem_list;
 		while(tmp) {
 			if (tmp->insn.id == insn->id) {
-				// found this instruction, so copy its mnemonic
-				(void)strncpy(insn->mnemonic, tmp->insn.mnemonic, sizeof(insn->mnemonic) - 1);
-				insn->mnemonic[sizeof(insn->mnemonic) - 1] = '\0';
+				char str[CS_MNEMONIC_SIZE];
```

Comment:
For simplicity i think this can be simplified with just one line calling snprintf. Most compilers will optimize the format string